cfgagent/intfconf.cpp

```diff
@@ -0,0 +1,126 @@
+#define LAG_PREFIX          "PortChannel"
+
+IntfConf::IntfConf(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, vector<string> tableNames) :
+        OrchBase(cfgDb, tableNames),
+        m_cfgIntfTable(cfgDb, CFG_INTF_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_cfgVlanIntfTable(cfgDb, CFG_VLAN_INTF_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_statePortTable(stateDb, STATE_PORT_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_stateLagTable(stateDb, STATE_LAG_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_stateVlanTable(stateDb, STATE_VLAN_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_appIntfTableProducer(appDb, APP_INTF_TABLE_NAME)
+{
+}
+
+void IntfConf::syncCfgDB()
+{
+    OrchBase::syncDB(CFG_INTF_TABLE_NAME, m_cfgIntfTable);
+    OrchBase::syncDB(CFG_VLAN_INTF_TABLE_NAME, m_cfgVlanIntfTable);
+}
+
+bool IntfConf::setIntfIp(string &alias, string &opCmd, string &ipPrefixStr)
+{
+    string cmd, res;
+
+    cmd = "ip address " + opCmd + " ";
+    cmd += ipPrefixStr + " dev " + alias;
+    swss::exec(cmd, res);
+    return true;
+}
+
+bool IntfConf::isIntfStateOk(string &alias)
+{
+    vector<FieldValueTuple> temp;
+
+    if (!alias.compare(0, strlen(VLAN_PREFIX), VLAN_PREFIX))
+    {
+        if (m_stateVlanTable.get(alias, temp))
+        {
+            SWSS_LOG_DEBUG("Port %s is ready\n", alias.c_str());
+            return true;
+        }
+    }
+    else if (!alias.compare(0, strlen(LAG_PREFIX), LAG_PREFIX))
+    {
+        if (m_stateLagTable.get(alias, temp))
+        {
+            SWSS_LOG_DEBUG("Lag %s is ready\n", alias.c_str());
+            return true;
+        }
+    }
+    else if (m_statePortTable.get(alias, temp))
+    {
+        SWSS_LOG_DEBUG("Port %s is ready\n", alias.c_str());
+        return true;
+    }
+
+    return false;
+}
+void IntfConf::doTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string keySeparator = CONFIGDB_KEY_SEPARATOR;
+        vector<string> keys = tokenize(kfvKey(t), keySeparator[0]);
+        string alias(keys[0]);
+
+        if (alias.compare(0, strlen(VLAN_PREFIX), VLAN_PREFIX))
+        {
+            /* handle IP over vlan Only for now, skip the rest */
+            it = consumer.m_toSync.erase(it);
+            continue;
+        }
+
+        size_t pos = kfvKey(t).find(CONFIGDB_KEY_SEPARATOR);
+        if (pos == string::npos)
+        {
+            SWSS_LOG_DEBUG("Invalid key %s", kfvKey(t).c_str());
+            it = consumer.m_toSync.erase(it);
+            continue;
+        }
+        IpPrefix ip_prefix(kfvKey(t).substr(pos+1));
+
+        SWSS_LOG_DEBUG("intfs doTask: %s", (dumpTuple(consumer, t)).c_str());
+
+        string op = kfvOp(t);
+        if (op == SET_COMMAND)
+        {
+            /* Don't proceed if port/lag/VLAN is not ready yet */
```

Comment:
add more comments here, retry. Later, we need to add subscriber for the state db to get notification when the port/lag/vlan is created.

Reply:
ok